src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -911,10 +911,19 @@ class spell_warr_slam : public SpellScriptLoader
             {
                 if (!GetHitUnit())
                     return;
-                GetCaster()->RemoveAurasDueToSpell(SPELL_WARRIOR_SLAM_AURA);
                 CastSpellExtraArgs args(TRIGGERED_FULL_MASK);
                 args.SpellValueOverrides.AddBP0(GetEffectValue());
                 GetCaster()->CastSpell(GetHitUnit(), SPELL_WARRIOR_SLAM, args);
+
+                Aura* aura = GetCaster()->GetOwnedAura(SPELL_WARRIOR_SLAM_AURA);
+                if (aura) {
```

Comment:
TC codestyle:
```cpp
                if (aura)
                {
                    uint8 auraCount = aura->GetCharges();
                    if (auraCount > 1)
                        aura->SetCharges(auraCount - 1);
                    else
                        GetCaster()->RemoveAurasDueToSpell(SPELL_WARRIOR_SLAM_AURA);
                }
```
(possibly with a blank line between 920 & 921 for readability)

Reply:
that won't make the code better

Reply:
@joschiwald All suggestions are welcome, how i can make it better ? cause i cant find anything to reduce aura one by one. anyway i'm going to fix this using spell_proc db